src/Evolution/Systems/ForceFree/Sources.cpp

```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/ForceFree/Sources.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace ForceFree {
+
+namespace detail {
+void sources_impl(
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*>
+        source_tilde_e,
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*>
+        source_tilde_b,
+    const gsl::not_null<Scalar<DataVector>*> source_tilde_psi,
+    const gsl::not_null<Scalar<DataVector>*> source_tilde_phi,
+
+    // temp variables
+    const tnsr::I<DataVector, 3, Frame::Inertial>&
+        trace_spatial_christoffel_second,
+
+    // EM args
+    const tnsr::I<DataVector, 3, Frame::Inertial>& tilde_e,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& tilde_b,
+    const Scalar<DataVector>& tilde_psi, const Scalar<DataVector>& tilde_phi,
+    const Scalar<DataVector>& tilde_q,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& spatial_current_density,
+    const double kappa_psi, const double kappa_phi,
+    // GR args
+    const Scalar<DataVector>& lapse,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& d_lapse,
+    const tnsr::iJ<DataVector, 3, Frame::Inertial>& d_shift,
+    const tnsr::II<DataVector, 3, Frame::Inertial>& inv_spatial_metric,
+    const Scalar<DataVector>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& extrinsic_curvature) {
+  // S(\tilde{E}^i)
+  raise_or_lower_index(source_tilde_e, d_lapse, inv_spatial_metric);
+  for (size_t i = 0; i < 3; ++i) {
+    source_tilde_e->get(i) -=
+        get(lapse) * trace_spatial_christoffel_second.get(i);
+    source_tilde_e->get(i) *= get(tilde_psi);
+    source_tilde_e->get(i) -= get(lapse) * get(sqrt_det_spatial_metric) *
+                              spatial_current_density.get(i);
```

Comment:
Is there a need to clarify (in the name) that this will only be the explicit part of the current?